### PR TITLE
Tickets/dm 38789

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

**Review by tribeiro — APPROVED** [review]
This looks ok. I left a few comments inline I hope you will consider before merging. 

A couple general comments. 

1. I noticed you are using WOUTER in some cases, possibly to indicate that this are log messages still in early debugging stages. I would reconsider using this approach. If you are looking to add a single string you can identify what the log is about, maybe come up with something that relates to the feature you are implementing something like "POWER MANAGEMENT" or "PM" to simplify. 
2. I would suggest you consider extracting the "power management" operation into its own class and then just using those operations here. I am not suggesting you do this now, in this PR, but for a future ticket (maybe before doing any further implementation). One of the advantages of this approach is that you would be able to test the power management feature without involving the CSC.

File: python/lsst/ts/mtdome/mtdome_csc.py [hunk 1]
```diff
@@ -519,6 +567,204 @@ async def end_enable(self, data: salobj.BaseDdsDataType) -> None:
...
+        params : `typing.Any`
+            The parameters to pass along with the command. This may be empty.
+        """
+        # All commands, that help reduce the power draw, are scheduled. It is
+        # up to the dome lower level control system to execute them or not.
+        if self.power_management_state == PowerManagementMode.NO_POWER_MANAGEMENT:
+            await self.write_then_read_reply(command=command, **params)
+        else:
+            scheduled_command = ScheduledCommand(command=command, params=params)
+            self.log.debug(f"WOUTER scheduling {scheduled_command=}")
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/mtdome/mtdome_csc.py:593]: Thanks for catching this and sorry for leaving those debug statements there. I removed all of them.

File: python/lsst/ts/mtdome/mtdome_csc.py [hunk 2]
```diff
@@ -519,6 +567,204 @@ async def end_enable(self, data: salobj.BaseDdsDataType) -> None:
...
+        depends on the expected power draw for the command and the available
+        power for the rotating part of the dome. If a command can be issued
+        then it is removed from the queue, otherwise not.
+        """
+        current_power_draw = await self._get_current_power_draw_for_llcs()
+        total_current_power_draw = sum([p for k, p in current_power_draw.items()])
+        power_available = (
+            AVAILABLE_CONTINUOUS_SLIP_RING_POWER_CAPACITY - total_current_power_draw
+        )
+        self.log.debug(f"WOUTER {current_power_draw=}, {power_available=}")
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/mtdome/mtdome_csc.py:610]: Thanks for catching this and sorry for leaving those debug statements there. I removed all of them.

File: python/lsst/ts/mtdome/mtdome_csc.py [hunk 3]
```diff
@@ -319,6 +349,14 @@ def __init__(
         # that have been sent.
         self.commands_without_reply: dict[int, CommandTime] = {}
 
+        # Queue of commands to execute. A Queue wouldn't work here since it
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/mtdome/mtdome_csc.py:352]: That would put it at the end (in case of a FIFO queue), right?

**Pull request merged**